### PR TITLE
fix: #4302 add wallet name

### LINKED ISSUE #4126: Total usd value of  account should be calculated based on the real time price of KSM/BSX/xxx
On the wallet hover popover, the total usd value of the current account should be calculated based on the real time price of KSM/BSX/xxx, according to the design.

image

image

_Originally posted by @Jarsen136 in https://github.com/kodadot/nft-gallery/issues/4063#issuecomment-1280157263_

### LINKED ISSUE #4302: Add name of wallet you are logged in and show assets of address
We should probably add the name of the account so users know under which name they are logged in.

@exezbcz will drop some ideas:)
image

### PR DESCRIPTION
**Pull request by petersopko**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).

👇 \_ Let's make a quick check before the contribution.

## PR Type

- [x] Bugfix
- [x] Feature
- [ ] Refactoring

## Context

- [x] Closes #4302, Closes #4126

#### Before submitting pull request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR

#### Optional

- [x] I've tested it at </bsx/collection>
- [x] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=FNyt7T1xdbhN7dagf7yGYCRJuE3R45VmTCE3tjzy1rKxa7y)

#### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

https://user-images.githubusercontent.com/73316633/206902254-9b400645-7642-436a-b3e0-7b34f1216594.mp4

### CODE REVIEW

**Comment by petersopko** [comment]
one thing left when it comes to functionality: changing wallet doesn't update the name in the dropdown, it updates only after refreshing the page. I've spent some time on how to do this and I have only stupid ideas, I'd appreciate some hints here 🥺

**Comment by petersopko** [comment]
@exezbcz btw, I have included your comments regarding the dividers etc., one question tho, shouldn't the blue color change on hover? I couldn't figure it out from the design.

**Comment by yangwao** [comment]
What about balance part?

image

**Comment by petersopko** [comment]
> What about balance part?
> 
> image

Must've missed that, I somehow didn't get this requirement from issue name / desc 😆 working on it.

**Comment by yangwao** [comment]
Anyway I've logged in, and this is not the name of my account :D 

image

**Comment by petersopko** [comment]
> Anyway I've logged in, and this is not the name of my account :D
> 
> image

oh okay, that's actually unrelated to this PR, also on beta, and caused by recent one here:
- https://github.com/kodadot/nft-gallery/pull/4467

@daiagi would you please create supplemental PR to your fix, so it doesn't break identity?

I've reverted it and it works, it seems that `identity` is getting passed as empty object from `IdentityIndex.vue`

image

**Comment by petersopko** [comment]
needs some more styling + storing basilisk/usd price (hopefully tomorrow evening).
image

@yangwao when it comes to USD price previews, basilisk is listed on coingecko, but not sure about tnkr, do we want to show that here? maybe hide tnkr for now ?

**Comment by yangwao** [comment]
ah yes if price is not available we will be showing iirc "--" in current design.

yes lot of tokens on lbp at some extent won't have USD price

**Comment by daiagi** [comment]
> > Anyway I've logged in, and this is not the name of my account :D
> > image
> 
> oh okay, that's actually unrelated to this PR, also on beta, and caused by recent one here:
> 
>     * [bug wrench make identityIndex respond to address change #4467](https://github.com/kodadot/nft-gallery/pull/4467)
> 
> 
> @daiagi would you please create supplemental PR to your fix, so it doesn't break identity?
> 
> I've reverted it and it works, it seems that `identity` is getting passed as empty object from `IdentityIndex.vue`
> 
> image



> > Anyway I've logged in, and this is not the name of my account :D
> > image
> 
> oh okay, that's actually unrelated to this PR, also on beta, and caused by recent one here:
> 
>     * [bug wrench make identityIndex respond to address change #4467](https://github.com/kodadot/nft-gallery/pull/4467)
> 
> 
> @daiagi would you please create supplemental PR to your fix, so it doesn't break identity?
> 


I cloned your branch and i'm not seeing what you are seeing
what i am seeing is that without my changes both the name **and** the address don't change until refresh
with my changes the adress changes when switching wallets

in both cases the name is wrong, it shows "My <extenstion name> wallet, and not the name i have given the wallets
and in both cases the name changes only with a refresh

> I've reverted it and it works, it seems that `identity` is getting passed as empty object from `IdentityIndex.vue`

FYI,i have chosen an initial value of empty object beacue that is the initial value in useIdentity
```ts
export default function useIdentity({ address, customNameOption }) {
  const { apiUrl } = useApi()
  const identity = ref<IdentityFields>({})
```

**Comment by petersopko** [comment]
@daiagi okay, I think we both got a bit confused here :D 

1. I need to start showing wallet name instead of extension name, somehow got that wrong. I will have time in the evening, I thought we want extension name. thank you for clarifying that for me 🚀 
2. My comment was regarding the on-chain identity, as I thought that's what was being discussed as `wallet name`. As an example, here's my [wallet on Subscan](https://kusama.subscan.io/account/GHoyYbdFWAsjACxgfAofb2M9KPARrbJCm1ZfZpzKyig9XRJ). https://github.com/kodadot/nft-gallery/pull/4467 broke that functionality, on-chain identity isn't shown anymore on beta, works for me again, after I reverted it. 

Here's what I'm seeing on beta:
image

Here it is still working on kodadot.xyz (or on my branch after reverting your PR): 
image



There should be `petersopko` instead of wallet address, you can see it working in my previous screenshots.

When it comes to reactivity in your PR, you're right, that part works, just the identity part needs to be fixed. 

So three different terms here, `extension name` (we don't want that), `custom wallet name` (we do want that, will get done in this PR), `on-chain identity`(needs fix after #4467).

You can set the `on-chain identity` here https://kodadot.xyz/identity

**Comment by yangwao** [comment]
I guess it's not worth showing assets the wallet doesn't have.
OR show them in greyscale? 
wyt @exezbcz  

image

**Comment by exezbcz** [comment]
> I guess it's not worth showing assets the wallet doesn't have. OR show them in greyscale? wyt @exezbcz
> 
> image
> 
> other than that, this is the very last PR I would like to squeeze into the upcoming release, so I would like to keep a steady 21 days cycle :)

i would not show it.

**Comment by yangwao** [comment]
let's refactor utils and let's go?

**Comment by petersopko** [comment]
> let's refactor utils and let's go?

working on it, for some reason I can't fetch my KSM balance on RMRK. Doesn't work on kodadot.xyz, beta or local, idk.

**Comment by petersopko** [comment]
currently testable at snek/bsx with various wallets.

todo:

- [ ] get ksm balance and show it in similar fashion. as I described, I'm currently not able to fetch KSM balance, it seems to be broken at prod and beta as well
- [ ] hide the whole asset section in case there are only 0 balance assets
- [ ] cleanup + style touchup

**Comment by daiagi** [comment]
@petersopko 
I've been exploring the issue you mentioned where my PR broke onchain identity and:
1. you were right, it does.
2. I have yet to understand why it breaks it or how to fix it

**Comment by petersopko** [comment]
> same here

I have solved issue with reactive wallet name by storing the wallet name in store, which is reactive when accessed through getters in components, i.e.
```
  get userWalletName(): string {
    return this.$store.getters['wallet/getWalletName']
  }
```

**Comment by yangwao** [comment]
why is there prefix `My` ?

image

> todo:

image

yes, we can do follow-up, 14 changed files are bit frightening to me 😅 this never went well with complex changes 



Otherwise is mergable I guess?

File: components/identity/module/IdentityLink.vue [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import { getExplorer } from '~/components/rmrk/Profile/utils'
+type Address = string | GenericAccountId | undefined
+
+const props = defineProps<{
+  shortenedAddress?: string | object
+  address?: Address
+}>()
+
+const { $store } = useNuxtApp()
+const { $buefy } = useNuxtApp()
```
Comment (roiLeo) [inline components/identity/module/IdentityLink.vue:31]:
```suggestion
const { $buefy, $store } = useNuxtApp()
```

File: components/navbar/ProfileDropdown.vue [hunk 2]
```diff
@@ -179,29 +179,17 @@
         <div class="has-text-grey is-size-7 mt-2">
           {{ $t('profileMenu.wallet') }}
         </div>
+        <span class="is-size-6">My {{ userWalletName }} wallet</span>
```
Comment (roiLeo) [inline components/navbar/ProfileDropdown.vue:182]: 😯

File: components/navbar/ProfileDropdown.vue [hunk 3]
```diff
@@ -179,29 +179,17 @@
...
         <Identity
           :address="account"
           class="navbar__address is-size-6"
           hide-identity-popover />
       </b-dropdown-item>
 
-      <hr class="dropdown-divider mx-4" aria-role="menuitem" />
```
Comment (roiLeo) [inline components/navbar/ProfileDropdown.vue:188]: I personally would not have moved `<hr />` tags to component

File: components/rmrk/Profile/ProfileAssetsList.vue [hunk 4]
```diff
@@ -0,0 +1,152 @@
+<template>
+  <div v-if="nonSnekBalance === 0">
+    <hr
+      class="profile-dropdown-divider dropdown-divider"
+      aria-role="menuitem" />
+    <table>
+      <thead>
+        <tr>
+          <th class="has-text-grey">Asset</th>
+          <th class="has-text-grey">Balance</th>
```
Comment (roiLeo) [inline components/rmrk/Profile/ProfileAssetsList.vue:10]:
```suggestion
          <th class="has-text-grey">{{ $t('assets') }}</th>
          <th class="has-text-grey">{{ $t('general.balance') }}</th>
```

Comment (roiLeo) [reply components/rmrk/Profile/ProfileAssetsList.vue:10]: would it be possible to move this file outside `/rmrk' folder?

File: components/identity/module/IdentityLink.vue (hunk 1 above)
Comment (yangwao) [reply components/identity/module/IdentityLink.vue:31]: legit

**Comment by yangwao** [comment]
> just `${walletName}`?

yes

File: locales/en.json [hunk 5]
```diff
@@ -1064,6 +1064,11 @@
     "disconnect": "Disconnect",
     "wallet": "Wallet"
   },
+  "profileAssetsList": {
+    "asset": "Asset",
+    "balance": "Balance",
+    "noAssets": "There are currently no assets in this wallet."
```
Comment (roiLeo) [inline locales/en.json:1070]: I would move asset & noAssets in 
```
"general": { ... }
```

**Comment by yangwao** [comment]
Seems https://github.com/kodadot/nft-gallery/pull/4476/files#diff-a4a33e7ecfb7969dbf41700112aa3cf6f2880fe4a397f3faececd4a001e67384 is copy of https://github.com/kodadot/nft-gallery/blob/main/components/bsx/Asset/AssetList.vue ?

I mean, I'm a bit against copying functions over file and duplicating code.

I'm merging it because this was the very last issue to have it released; otherwise, I dislike the code approach ignoring DRY approach.

pay 60 USD

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $60 @ 29.12 USD/KSM ~ 2.06 $KSM
🧗 [FNyt7T1xdbhN7dagf7yGYCRJuE3R45VmTCE3tjzy1rKxa7y](https://kusama.subscan.io/account/FNyt7T1xdbhN7dagf7yGYCRJuE3R45VmTCE3tjzy1rKxa7y)
🔗 [0x190ecd6efea6a858e7132ad5d3c3418a91daa43f1824e71ca3ef9d1d61d66f9b](https://kusama.subscan.io/extrinsic/0x190ecd6efea6a858e7132ad5d3c3418a91daa43f1824e71ca3ef9d1d61d66f9b)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Comment by petersopko** [comment]
> dislike the code approach ignoring DRY approach
> 
> pay 60 USD

I'm not too fond of the state in which it got merged in either, and I wish I had time to clean up/finish. Maybe next time :)

**Pull request merged**